src/System.Management.Automation/engine/remoting/fanin/InitialSessionStateProvider.cs

```diff
@@ -2890,4 +2927,110 @@ internal static string[] TryGetStringArray(object hashObj)
         }
     }
     #endregion
+
+    #region DISCFileValidation
+
+    internal static class DISCFileValidation
+    {
+        // Set of supported configuration options for a PowerShell InitialSessionState.
+#if UNIX
```

Comment:
It looks to me the difference between Unix and Windows is just `ExecutionPolicy`. How about does the initialization in a static constructor for the commonly supported options, and then do `SupportedConfigOptions.Add('ExecutionPolicy')` for windows only?

Reply:
I think have the common ones created in a `Hashset` and add platform unique ones to the `Hashset` instance separately is more manageable in the long run. It's hard to tell what the differences are by having 2 lists.

Reply:
Fair enough.  Agree to disagree.